src/dyncfg-ld/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+            Err(TimeoutError::DeadlineElapsed) => {
+                tracing::info!("SyncedConfigSet initialize on boot: initialize has timed out");
+                None
+            }
+        }
+    } else {
+        None
+    };
+
+    let synced = SyncedConfigSet {
+        set,
+        ld_client,
+        ld_ctx: ld_ctx(build_info, ctx_builder)?,
+    };
+    synced.sync()?;
+    task::spawn(
+        || "SyncedConfigSet sync_loop",
+        synced.sync_loop(config_sync_loop_interval),
+    );
+    Ok(())
+}
+
+fn ld_ctx<F>(build_info: &'static BuildInfo, ctx_builder: F) -> Result<ld::Context, anyhow::Error>
+where
+    F: FnOnce(&mut ld::MultiContextBuilder),
+{
+    // Register multiple contexts for this client.
+    //
+    // Unfortunately, it seems that the order in which conflicting targeting
+    // rules are applied depends on the definition order of feature flag
+    // variations rather than on the order in which context are registered with
+    // the multi-context builder.
+    let mut builder = ld::MultiContextBuilder::new();
+
+    builder.add_context(
+        ld::ContextBuilder::new(build_info.sha)
+            .kind("build")
+            .set_string("semver_version", build_info.semver_version().to_string())
+            .build()
+            .map_err(|e| anyhow::anyhow!(e))?,
+    );
+
+    ctx_builder(&mut builder);
+
+    builder.build().map_err(|e| anyhow::anyhow!(e))
+}
+
+struct SyncedConfigSet {
+    set: ConfigSet,
+    ld_client: Option<ld::Client>,
+    ld_ctx: ld::Context,
+}
+
+impl SyncedConfigSet {
+    /// Returns a future that periodically polls LaunchDarkly and updates the ConfigSet.
+    async fn sync_loop(self, tick_interval: Option<Duration>) {
+        let Some(tick_interval) = tick_interval else {
+            tracing::info!("skipping SyncedConfigSet sync as tick_interval = None");
+            return;
+        };
+
+        let mut interval = time::interval(tick_interval);
+        interval.set_missed_tick_behavior(time::MissedTickBehavior::Skip);
+
+        tracing::info!(
+            "synchronizing SyncedConfigSet values every {} seconds",
+            tick_interval.as_secs()
+        );
+
+        loop {
+            interval.tick().await;
+
+            if let Err(err) = self.sync() {
+                tracing::info!("SyncedConfigSet: {err}");
+            }
+        }
+    }
+
+    /// Reads current values from LaunchDarkly and updates the ConfigSet.
+    fn sync(&self) -> Result<ConfigUpdates, anyhow::Error> {
+        let mut updates = ConfigUpdates::default();
+        let Some(ld_client) = &self.ld_client else {
+            return Ok(updates);
+        };
+        for entry in self.set.entries() {
+            let val = dyn_into_flag(entry.val()).expect("new() verifies all configs can convert");
+            let flag_var = ld_client.variation(&self.ld_ctx, entry.name(), val);
+            // TODO: Is there a way to avoid this on every loop if the flag didn't change?
```

Comment:
Could store a map of name -> ld::FlagValue, update it at the end of a sync, and use that here to diff what's changed

Reply:
I looked at the adapter code again and it seems that it updates everything each time. I'm going to leave as is.